velox/functions/prestosql/window/tests/SimpleAggregatesTest.cpp

```diff
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/window/tests/WindowTestBase.h"
+
+using namespace facebook::velox::exec::test;
+
+namespace facebook::velox::window::test {
+
+namespace {
+
+class SimpleAggregatesTest : public WindowTestBase {
+ protected:
+  explicit SimpleAggregatesTest(const std::string& function)
+      : function_(function) {}
+
+  void testTwoColumnOverClauses(const RowVectorPtr& vectors) {
+    WindowTestBase::testTwoColumnOverClauses({vectors}, function_);
+  }
+
+  void testWindowFunction(
+      const std::vector<RowVectorPtr>& vectors,
+      const std::vector<std::string>& overClauses) {
+    WindowTestBase::testWindowFunction(vectors, function_, overClauses);
+  }
+
+  const std::string function_;
+};
+
+class MultiAggregatesTest : public SimpleAggregatesTest,
+                            public testing::WithParamInterface<std::string> {
+ public:
+  MultiAggregatesTest() : SimpleAggregatesTest(GetParam()) {}
+};
+
+TEST_P(MultiAggregatesTest, basic) {
+  vector_size_t size = 10;
+
+  auto vectors = makeRowVector({
+      makeFlatVector<int32_t>(
+          size, [](auto row) -> int32_t { return row % 10; }),
+      makeFlatVector<int32_t>(
+          size, [](auto row) -> int32_t { return row % 7; }),
+      makeFlatVector<int32_t>(size, [](auto row) -> int32_t { return row; }),
+  });
+
+  testTwoColumnOverClauses({vectors});
+}
+
+TEST_P(MultiAggregatesTest, singlePartition) {
+  // Test all input rows in a single partition.
+  vector_size_t size = 1'000;
+
+  auto vectors = makeRowVector({
+      makeFlatVector<int32_t>(size, [](auto /* row */) { return 1; }),
+      makeFlatVector<int32_t>(size, [](auto row) { return row % 50; }),
+      makeFlatVector<int32_t>(size, [](auto row) -> int32_t { return row; }),
+  });
+
+  testTwoColumnOverClauses({vectors});
+}
+
+TEST_P(MultiAggregatesTest, randomInput) {
+  auto vectors = makeVectors(
+      ROW({"c0", "c1", "c2", "c3"},
+          {BIGINT(), SMALLINT(), INTEGER(), BIGINT()}),
+      10,
+      2);
+  createDuckDbTable(vectors);
+
+  std::vector<std::string> overClauses = {
+      "partition by c0 order by c1, c2, c3",
+      "partition by c1 order by c0, c2, c3",
+      "partition by c0 order by c1 desc, c2, c3",
+      "partition by c1 order by c0 desc, c2, c3",
+      "partition by c0 order by c1",
+      "partition by c0 order by c2",
+      "partition by c0 order by c3",
+      "partition by c1 order by c0 desc",
+      "partition by c0, c1 order by c2, c3",
+      "partition by c0, c1 order by c2",
+      "partition by c0, c1 order by c2 desc",
+      "order by c0, c1, c2, c3",
+      "partition by c0, c1, c2, c3",
+  };
+
+  testWindowFunction(vectors, overClauses);
+}
+
+VELOX_INSTANTIATE_TEST_SUITE_P(
+    SimpleAggregatesTest,
+    MultiAggregatesTest,
+    testing::ValuesIn({
+        std::string("sum(c2)"),
```

Comment:
Would you add tests for aggregate function that use non-fixed-with accumulators? For example, array_agg or min/max over string column?

Reply:
Done in https://github.com/facebookincubator/velox/pull/3093